packages/ingest/granule.js

```diff
@@ -464,19 +470,60 @@ class Granule {
     return [null, null];
   }
 
+  /**
+   * rename s3 file with timestamp
+   *
+   * @param {string} bucket - bucket of the file
+   * @param {string} key - s3 key of the file
+   * @returns {Promise} promise that resolves when file is renamed
+   */
+  async renameS3FileWithTimestamp(bucket, key) {
+    const formatString = 'YYYYMMDDTHHmmssSSS';
+    let renamedKey = `${key}.v${moment.utc().format(formatString)}`;
+
+    // if the renamed file already exists, get a new name
+    // eslint-disable-next-line no-await-in-loop
+    while (await aws.s3ObjectExists({ Bucket: bucket, Key: renamedKey })) {
+      renamedKey = `${key}.v${moment.utc().format(formatString)}`;
```

Comment:
Might just want to put a log message in here. This case seems like it should never really get hit, right?

Reply:
Removed the check.  I was using different format at first.
Updated to use the timestamp from the file.